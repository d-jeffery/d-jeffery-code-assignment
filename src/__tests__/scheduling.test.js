const { Scheduling } = require('../scheduling');
const { Client } = require('../client');
const { Provider } = require('../provider');

describe('Scheduling', () => {
  test('register provider', () => {
    const scheduling = new Scheduling();
    scheduling.registerProvider(new Provider('Dr Jones'));

    expect(scheduling).toMatchSnapshot();
  });

  test('register client', () => {
    const scheduling = new Scheduling();
    scheduling.registerClient(new Client('Mr Harrison'));

    expect(scheduling).toMatchSnapshot();
  });

  test('registerBooking', () => {
    const scheduling = new Scheduling();
    const provider = new Provider('Dr Jones');
    const client = new Client('Mr Harrison');

    scheduling.registerProvider(provider);
    scheduling.registerClient(client);

    provider.addAppointments(1723680000000, 1723680000000 + (60 * 60 * 1000));

    const token = scheduling.registerBooking(client, provider, 1723680000000);

    expect(scheduling.unconfirmedAppointments.get(token)).toMatchSnapshot();
  });

  test('confirmBooking', () => {
    const scheduling = new Scheduling();
    const provider = new Provider('Dr Jones');
    const client = new Client('Mr Harrison');

    scheduling.registerProvider(provider);
    scheduling.registerClient(client);

    provider.addAppointments(1723680000000, 1723680000000 + (60 * 60 * 1000));

    const token = scheduling.registerBooking(client, provider, 1723680000000);
    scheduling.confirmBooking(token);

    expect(scheduling.unconfirmedAppointments.size).toBe(0);
    expect(client.getBookings()).toMatchSnapshot();
    expect(provider.getAppointments()).toMatchSnapshot();
  });

    test('cancelBooking', () => {
        const scheduling = new Scheduling();
        const provider = new Provider('Dr Jones');
        const client = new Client('Mr Harrison');

        scheduling.registerProvider(provider);
        scheduling.registerClient(client);

        provider.addAppointments(1723680000000, 1723680000000 + (60 * 60 * 1000));

        const token = scheduling.registerBooking(client, provider, 1723680000000);
        scheduling.cancelBooking(token);

        expect(scheduling.unconfirmedAppointments.size).toBe(0);
        expect(client.getBookings()).toMatchSnapshot();
        expect(provider.getAppointments()).toMatchSnapshot();
    });
});
