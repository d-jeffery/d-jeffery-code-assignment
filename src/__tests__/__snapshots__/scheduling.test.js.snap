// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scheduling cancelBooking 1`] = `Map {}`;

exports[`Scheduling cancelBooking 2`] = `
Map {
  1723680000000 => false,
  1723680900000 => false,
  1723681800000 => false,
  1723682700000 => false,
}
`;

exports[`Scheduling confirmBooking 1`] = `
Map {
  1723680000000 => "Dr Jones",
}
`;

exports[`Scheduling confirmBooking 2`] = `
Map {
  1723680000000 => true,
  1723680900000 => false,
  1723681800000 => false,
  1723682700000 => false,
}
`;

exports[`Scheduling register client 1`] = `
Scheduling {
  "clients": [
    Client {
      "bookings": Map {},
      "name": "Mr Harrison",
    },
  ],
  "providers": [],
  "unconfirmedAppointments": Map {},
}
`;

exports[`Scheduling register provider 1`] = `
Scheduling {
  "clients": [],
  "providers": [
    Provider {
      "appointments": Map {},
      "name": "Dr Jones",
    },
  ],
  "unconfirmedAppointments": Map {},
}
`;

exports[`Scheduling registerBooking 1`] = `
{
  "client": "Mr Harrison",
  "provider": "Dr Jones",
  "slot": 1723680000000,
}
`;
